packages/gestalt/src/HelpButton.js

```diff
@@ -0,0 +1,178 @@
+   */
+  link?: {|
+    accessibilityLabel?: string,
+    href: string,
+    onClick?: ({|
+      event: SyntheticMouseEvent<HTMLAnchorElement> | SyntheticKeyboardEvent<HTMLAnchorElement>,
+      dangerouslyDisableOnNavigation: () => void,
+    |}) => void,
+    text?: string,
+  |},
+|};
+
+const DEFAULT_ZINDEX = 2;
+
+/**
+ * [HelpButton](https://gestalt.pinterest.systems/web/helpbutton) is an affordance that accompanies an element on the screen. It helps describe or provide assistance on how to use the accompanying element.
+ */
+export default function HelpButton({
+  accessibilityPopoverLabel,
+  idealDirection = 'down',
+  link,
+  text,
+  zIndex,
+}: Props): Node {
+  const ref = useRef(null);
+  const [isOpen, setIsOpen] = useState(false);
+  const [isHover, setIsHover] = useState(false);
+  const {
+    accessibilityTooltipMessage,
+    accessibilityLinkLabel: accessibilityDefaultLinkLabel,
+    accessibilityIcon,
+  } = useDefaultLabelContext('HelpButton');
+
+  useEffect(() => {
+    if (isOpen && ref.current) {
+      ref.current.focus();
+    }
+  }, [isOpen]);
+
+  const toggleView = () => {
+    setIsOpen((currVal) => !currVal);
+  };
+
+  const bgIconColor = isHover ? 'default' : 'subtle';
+
+  const tooltipZIndex = zIndex ?? new FixedZIndex(DEFAULT_ZINDEX - 1);
+
+  return (
+    <Fragment>
+      <Box dangerouslySetInlineStyle={{ __style: { display: 'inline-block' } }} tabIndex={-1}>
+        <TapArea
+          accessibilityExpanded={isOpen}
+          accessibilityControls="info-dialog"
+          accessibilityLabel={accessibilityTooltipMessage}
+          fullWidth={false}
+          onTap={toggleView}
+          ref={ref}
+          role="button"
+          rounding="circle"
+          onMouseEnter={() => setIsHover(true)}
+          onMouseLeave={() => setIsHover(false)}
+        >
+          <Tooltip
+            text={accessibilityTooltipMessage}
+            idealDirection={idealDirection}
+            zIndex={tooltipZIndex}
+          >
+            <Box
+              width={16}
+              height={16}
+              rounding="circle"
+              display="flex"
+              alignItems="center"
+              justifyContent="center"
+              dangerouslySetInlineStyle={{
+                __style: {
+                  backgroundColor: `var(--color-text-${bgIconColor})`,
+                },
+              }}
+            >
+              <Icon
+                accessibilityLabel={accessibilityIcon}
+                color="inverse"
+                icon="question-mark"
+                size={8}
+              />
+            </Box>
+          </Tooltip>
+        </TapArea>
+      </Box>
+      {isOpen && (
+        <Layer
+          zIndex={
+            zIndex
+              ? new CompositeZIndex([new FixedZIndex(zIndex.index())])
```

Comment:
`new FixedZIndex(zIndex.index())`

Why is this necessary? Can we not use `zIndex` directly?

Reply:
We did some flow type mistake, I follow the current docs reference to type the zIndex and when I use the prop typed like `Indexable` on `new compositeZindex` the error below is showed:

![Screenshot 2023-02-07 at 11 10 12](https://user-images.githubusercontent.com/16820917/217275208-a2050255-a732-4c74-a79d-071f7cf1f567.png)

Reply:
That's really odd. FixedZIndex and CompositeZIndex _are_ Indexable: https://github.com/pinterest/gestalt/blob/b29dd46b656c4cbe0b1343dad1a81fe88f3462b5/packages/gestalt/src/zIndex.js

@jackhsu978 do you have any tips here? I'm confused why Flow is unhappy with this usage

Reply:
The line https://github.com/pinterest/gestalt/blob/b29dd46b656c4cbe0b1343dad1a81fe88f3462b5/packages/gestalt/src/zIndex.js#L36 is the "blocker", because the constructor receives an array of ZIndex or CompositeZIndex, and not a Array of Indexable, as we expect.